examples/medplum-task-demo/src/App.tsx

```diff
@@ -1,56 +1,123 @@
-import { Anchor, AppShell, Button, Group, Loader, Text } from '@mantine/core';
-import { ErrorBoundary, Logo, useMedplum } from '@medplum/react';
-import { Suspense } from 'react';
-import { Link, Route, Routes } from 'react-router-dom';
-import { HomePage } from './HomePage';
-import { LandingPage } from './LandingPage';
-import { ResourcePage } from './ResourcePage';
-import { SignInPage } from './SignInPage';
+import {
+  Operator,
+  SearchRequest,
+  formatCodeableConcept,
+  formatSearchQuery,
+  getReferenceString,
+  normalizeErrorString,
+} from '@medplum/core';
+import { PractitionerRole } from '@medplum/fhirtypes';
+import { AppShell, Loading, Logo, NavbarLink, useMedplum, useMedplumProfile } from '@medplum/react';
+import { IconCategory, IconDatabaseImport, IconFileImport, IconGridDots, IconUser } from '@tabler/icons-react';
+import { Suspense, useEffect, useState } from 'react';
+import { Route, Routes } from 'react-router-dom';
+import { LandingPage } from './pages/LandingPage';
+import { ResourcePage } from './pages/ResourcePage';
+import { SearchPage } from './pages/SearchPage';
+import { SignInPage } from './pages/SignInPage';
+import { TaskPage } from './pages/TaskPage';
+import { UploadDataPage } from './pages/UploadDataPage';
 
 export function App(): JSX.Element | null {
   const medplum = useMedplum();
+  const profile = useMedplumProfile();
+
+  const profileReference = profile && getReferenceString(profile);
+  const [userLinks, setUserLinks] = useState<NavbarLink[]>([
+    { icon: <IconGridDots />, label: 'All Tasks', href: '/Task' },
+  ]);
+
+  useEffect(() => {
+    if (!profileReference) {
+      return;
+    }
+    const myTasksQuery = formatSearchQuery({
+      resourceType: 'Task',
+      fields: ['code', '_lastUpdated', 'owner', 'for', 'priority'],
+      sortRules: [{ code: '-priority-order,due-date' }],
+      filters: [
+        { code: 'owner', operator: Operator.EQUALS, value: profileReference },
+        { code: 'status:not', operator: Operator.EQUALS, value: 'completed' },
+      ],
+    });
+
+    const myTasksLink = { icon: <IconCategory />, label: 'My Tasks', href: `/Task${myTasksQuery}` };
+
+    async function getUserPractitionerRoles(): Promise<PractitionerRole[]> {
```

Comment:
It would be cleaner if this was not a function - the function complexity is due to the check for `if (profileReference)` which is already checked above.  `searchResources` will always return an array, so the `?? []` is unnecessary on line 52